datachannel.md

```diff
@@ -0,0 +1,171 @@
+# Data Channel
+
+[Data Channel](https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13)
+is a non-media transport protocol in WebRTC framework, designed for exchanging
+data from peer to peer. It advertise the following benefits:
+
+- Implemented in major browsers (except Safari)
+- Support heterogeneous network environment e.g. NAT, firewall.
+- Easily support media transport for remote playback
+- Data framing format is already defined
+
+# Design and Specification
+
+- [WebRTC overview](https://tools.ietf.org/html/draft-ietf-rtcweb-overview-18)
+- IETF draft: [WebRTC Data Channel Establishment Protocol](https://tools.ietf.org/html/draft-ietf-rtcweb-data-protocol-09)
+- IETF draft: [Data Channel](https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13)
+
+# Presentation API functionality
+
+Data Channel is designed to be a non-media transport for WebRTC, with built-in
+sub protocol negociation mechanism defined.
+
+A Data Channel is built on top of SCTP/DTLS/UDP protocol stack, which supports
+reliable data transmition. Each Data Cahnnel has a unique integer ID and a label.
+Mutiple Data Channels can share the same DTLS connection.
+
+Assume that a single Data Channel is used as a control channel to transmit control
+messages between the controller and receiver and one Data Channel is created for each
+connected Presentation Connection for trasmitting application data.
+
+An extra signalling channel and protocol for exchanging ICE candidates is required
+to bootstrap the control channel. The bootstraping procedure needs to be executed
+for the first connecting to that receiver before the Presentation Initiation.
+
+## Presentation Initiation
+
+1. Send a control message on control channel from the controller to the display to
+   start the presentation with ID and URL for the initial presentation connection.
+2. Receiver spawn a Data Channel on the same RTC connection of control channel, with
+   presentation ID as label.
+3. Get a reply on control channel with the ID and URL to confirm success
+   (or report an error).
+
+## Presentation Resumption
+
+1. Send a control message on control channel from the controller to the display
+   reconnect to a presentation with ID and URL for the initial presentation connection.
+2. Receiver spawn a Data Channel on the same RTC connection of control channel, with
```

Comment:
If a second controller wants to initiate connection to the same presentation, would it have a duplicate Data Channel label?